R/count_teleconnections.R

```diff
@@ -149,21 +150,42 @@ count_watershed_teleconnections <- function(data_dir,
           .[["GCAM_ID"]] %>% unique() %>%
           length() -> tc_n_cropcover
 
-        # filter out where "is crop" is false and count land classes, adding 1 class to account for agriculture.
-        crop_and_landcover_types %>%
-          filter(is_crop == FALSE) %>%
-          .[["CDL_Class"]] %>% unique() %>%
-          length() -> number_landclasses_ex_ag
-
-        if (tc_n_cropcover > 0){
-          tc_n_landclasses = number_landclasses_ex_ag + 1
-        }else{
-          tc_n_landclasses = number_landclasses_ex_ag
-        }
         # TELECONNECTION - NUMBER OF FLOOD CONTROL DAMS WITHIN WATERSHED.
         flood_control_dams[watersheds_city, ] %>%
           length() -> tc_fcdam
 
+        # TELECONNECTION - RANK WATERSHED BASED ON % OF DEVELOPED/CULTIVATED AREA.
+        # Get frequencies of each type of land cover within the designated watershed.
+        mask_raster_to_polygon(cropcover_USA, watersheds_city) %>%
+          raster::freq() %>%
+          as.data.frame() -> freqdf
+        # Remove NA and all categories that are not land cover/use(water/background).
+        freqdf[!is.na(freqdf$value),] %>%
+          filter(!value %in% c(0,81,83,88,111)) -> all_land
+        # New df with only crops and developement categories
+        freqdf[!is.na(freqdf$value),] %>%
+          filter(!value %in% c(0,63:65,81,83:92,111,112,131:195)) -> dev_and_crop
+        # Add cell count for all the land to get total land coverage.
+        totcells <- sum(all_land$count)
+        # Add cell count for all development and crop counts.
+        totaldevcrop <- sum(dev_and_crop$count)
+        # Find percent area for development and crops.
+        percent.area <- 100*totaldevcrop/totcells
+        # Assign to category based on percent area.
+        if(percent.area <= 1){
+          watershed_condition <- "Very Low"
+        }else if(percent.area > 1 & percent.area <= 5){
+          watershed_condition <- "Low"
+        }else if(percent.area > 5 & percent.area <= 15){
+          watershed_condition <- "Average"
+        }else if(percent.area > 15 & percent.area <= 30){
+          watershed_condition <- "High"
+        }else if(percent.area > 30){
+          watershed_condition <- "Very High"
+        }else{
+          message("No Watershed Detected.")
```

Comment:
Not sure this is necessary as cases with no watershed are removed at an earlier stage.